Lib/idlelib/pyshell.py

```diff
@@ -1445,11 +1451,16 @@ def main():
         dir = os.getcwd()
         if dir not in sys.path:
             sys.path.insert(0, dir)
-    # check the IDLE settings configuration (but command line overrides)
+    # Check the IDLE settings configuration (but command line overrides).
     edit_start = idleConf.GetOption('main', 'General',
                                     'editor-on-startup', type='bool')
     enable_edit = enable_edit or edit_start
     enable_shell = enable_shell or not enable_edit
+    if idleConf.GetOption('main', 'ShellWindow',
+                          'startup-code-on', type='bool'):
+        config_startup_code = idleConf.GetOption(
+                'main', 'ShellWindow', 'shell-startup-code')
```

Comment:
I seems to have missed the difference between 'restart-code-on' and 'startup-code-on'.  I need to look at pyshell changes, especially, in broader context.  I want to understand and document the pyshell code better,  with comments and docstrings , before changing it.

Reply:
The 'restart-code-on' and 'startup-code-on' are checkbuttons that indicate when to run the code defined within config.  Although your comment on the ticket mentions just having one code set for it, since startup and restart are different now, I added the checkbutton so that the code could be run just at initial startup, just at restart (Ctrl-F6), or both.

I thought documenting and tests for pyshell might be added as a dependency, just as the editor tests project would be a dependency.